presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveRedundantAggregateDistinct.java

```diff
@@ -61,16 +61,18 @@ public Result apply(AggregationNode node, Captures captures, Context context)
         ImmutableMap.Builder<VariableReferenceExpression, AggregationNode.Aggregation> aggregationsBuilder = ImmutableMap.builder();
 
         for (Map.Entry<VariableReferenceExpression, AggregationNode.Aggregation> agg : node.getAggregations().entrySet()) {
-            Set<VariableReferenceExpression> varAndGroupingKeySet =
-                    Stream.concat(node.getGroupingKeys().stream().map(VariableReferenceExpression.class::cast),
-                                    (agg.getValue()).getArguments().stream().map(VariableReferenceExpression.class::cast))
-                            .collect(Collectors.toSet());
-            if (agg.getValue().isDistinct() && ((GroupReference) node.getSource()).getLogicalProperties().get().isDistinct(varAndGroupingKeySet)) {
-                aggregationsBuilder.put(agg.getKey(), removeDistinct(agg.getValue()));
-            }
-            else {
-                aggregationsBuilder.put(agg);
+            if (node.getGroupingKeys().stream().allMatch(key -> key instanceof VariableReferenceExpression) &&
```

Comment:
I know most of this code was here before but perhaps refactor that as functions to increase readability?